pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/MeasurementsKokkos.hpp

```diff
@@ -506,11 +507,18 @@ class Measurements final
      * @return Floating point std::vector with probabilities.
      * The basis columns are rearranged according to wires.
      */
-    std::vector<PrecisionT> probs(const std::vector<std::size_t> &wires) {
+    auto probs(const std::vector<std::size_t> &wires)
+        -> std::vector<PrecisionT> {
         PL_ABORT_IF_NOT(
             std::is_sorted(wires.cbegin(), wires.cend()),
             "LightningKokkos does not currently support out-of-order wire "
             "indices with probability calculations");
+
+        // If all wires are requested, dispatch to `this->probs()`
+        if (wires.size() == this->_statevector.getNumQubits()) {
+            return this->probs();
+        }
+
```

Comment:
We have proper [Python bindings](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/core/src/bindings/Bindings.hpp#L439-L447) for probs(). The problem is that we always provide wires in the devices' class even if this is an all-wires situation. The more straightforward solution would be to have a test somewhere (like you did here) and redirect when possible. The best place for that is in the C++ layer. Also, this is a cheap check.